src/DIRAC/WorkloadManagementSystem/JobWrapper/Watchdog.py

```diff
@@ -974,10 +974,17 @@ def getNodeInformation(self):
         result["Memory(kB)"] = int(psutil.virtual_memory()[1] / 1024)
         result["LocalAccount"] = getpass.getuser()
 
-        with open("/proc/cpuinfo", "r") as cpuinfo:
-            info = cpuinfo.readlines()
-        result["ModelName"] = info[4].split(":")[1].replace(" ", "").replace("\n", "")
-        result["CacheSize(kB)"] = [x.strip().split(":")[1] for x in info if "cache size" in x][0].strip()
+        if os.path.exists("/proc/cpuinfo"):
```

Comment:
What about doing something like this rather than assuming the layout of the file:

```python
In [1]: from pathlib import Path
   ...:
   ...: result = {}
   ...:
   ...: path = Path("/proc/cpuinfo")
   ...: if path.is_file():
   ...:     # Assume all of the CPUs are the same and only consider the first one
   ...:     raw_info = path.read_text().split("\n\n")[0]
   ...:     info = dict(map(str.strip, x.split(":", 1)) for x in raw_info.split("\n"))
   ...:     result["ModelName"] = info.get("model name", "Unknown")
   ...:     result["CacheSize(kB)"] = info.get("cache size", "Unknown")
   ...:
   ...: result
Out[1]: {'ModelName': 'AMD EPYC 7552 48-Core Processor', 'CacheSize(kB)': '512 KB'}
```

This is slightly different to the current output which has the spaces removed from the model name but I see no reason for that:

```
{'ModelName': 'AMDEPYC755248-CoreProcessor', 'CacheSize(kB)': '512 KB'}
```

Reply:
`pathlib` is only from python3.4 and here we still need to keep the compatibility with python2.

Reply:
The use of `pathlib` isn't relavent, it's mostly about `info = dict(map(str.strip, x.split(":", 1)) for x in raw_info.split("\n"))` rather than assuning the line numbers

Reply:
I promise I will do it properly in integration branch, let's just fix the bug now.